### PR TITLE
Added Viktor's doc

### PR DESCRIPTION
**Pull request by Dainternetdude**
I copy-pasted @viktor40's ["Minecraft Deep Dive" gdoc](https://docs.google.com/document/d/1dkCww2bGPWUvtN6lDG8Lsh97NCVeGYBC6ldr7lbuW4U/edit) with details on server tick, mob spawning, and weather. Then I added the markdown formatting. Then I made some readability edits, did some proofreading, and updated some parts that I knew were outdated.

### CODE REVIEW

File: GameTick/Weather.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+Rain will only begin to fall when the gradient is bigger than 0.2 and will only stop falling when it’s smaller than 0.2 again. In this case we’re speaking visually of course. Internally the properties are already set when the gradients start changing.
+
+## Sleeping
+Next in this part of the tick sleeping happens. First the time of day will be set to morning if a player has just finished sleeping. It will also reset the weather if the weather cycle is enabled. Resetting the weather set `rainTime` and `thunderTime` to 0 and `raining` and `thundering` to false.
+
+## Daylight Cycle
+In this part, if daylight cycle is enabled, the world time will be increased by one & the time of day will change.
+
+## What about lightning strikes?
+Lightning strikes don’t happen in this part of the tick but rather in the chunk part, which is the next part we’ll cover.
```
Comment (JoakimThorsen) [inline GameTick/Weather.md]: "which is the next part we'll cover" is residue for this being part of the game tick page. Should be rephrased.

File: GameMechanics/MobSpawning.md [hunk 2]
```diff
@@ -3,19 +3,107 @@ title: Spawning algorithm
...
-most monsters have a maximum pack size of 4
-### Pack Spawn Attempt Jumping
-spawn attempts that begin in one location, may jump to other locations via the pack spawning mechanic, even if the original location was not valid. this is why u dont want to have conductive blocks near the foot level of spawn pads in mobfarms (i think)
-### Mob Caps:TODO link
\ No newline at end of file
+
+# Valid Spawning Locations
+
+## Spawnable blocks
+Mobs can spawn on blocks with a solid top surface.
```
Comment (JoakimThorsen) [inline GameMechanics/MobSpawning.md]: There are a few exceptions to this.

File: GameMechanics/MobSpawning.md [hunk 3]
```diff
@@ -3,19 +3,107 @@ title: Spawning algorithm
...
+
+## Spawning inside other entities
+up to 1.8 monsters can't spawn inside other entities. then up to 1.14 (someone confirm this) they can again. currently they can't again. (please confirm)
```
Comment (JoakimThorsen) [inline GameMechanics/MobSpawning.md]: lol (bruh (please confirm))

File: GameMechanics/MobSpawning.md [hunk 4]
```diff
@@ -3,19 +3,107 @@ title: Spawning algorithm
...
+## Spawn Groups:
+`SpawnGroup(String name, int spawnCap, boolean peaceful, boolean isAnimal, int despawnRange)`
+```java
+MONSTER("monster", 70, false, false, 128),
+CREATURE("creature", 10, true, true, 128),
+AMBIENT("ambient", 15, true, false, 128),
+WATER_CREATURE("water_creature", 5, true, false, 128),
+WATER_AMBIENT("water_ambient", 20, true, false, 64),
+MISC("misc", -1, true, true, 128);
+```
```
Comment (JoakimThorsen) [inline GameMechanics/MobSpawning.md]: this could be a table

File: GameTick.md [hunk 5]
```diff
@@ -18,118 +19,138 @@ The ones commonly used for redstone are these:
...
+`net.minecraft.world.border.WorldBorder.tick`  
+Called in the game loop.
+
+The only thing that will happen during this "ticking" of the world border is checking if the world border needs to change in size, and executing that.
+
+# [Weather Cycle](https://techmcdocs.github.io/pages/GameTick/Weather/)
+See separate article.
+
+# [Block & Fluid Ticks (Tile Tick Phase)](https://techmcdocs.github.io/pages/GameTick/BlockTicks/)
+See separate article.
```
Comment (JoakimThorsen) [inline GameTick.md]: I would change it to mimic the minecraft wiki:
```
## Weather Cycle
See separate article: [Weather](https://techmcdocs.github.io/pages/GameTick/Weather/)

## Block & Fluid Ticks (Tile Tick Phase)
See separate article: [BlockTicks](https://techmcdocs.github.io/pages/GameTick/BlockTicks/)
```

File: GameTick.md [hunk 6]
```diff
@@ -143,26 +164,320 @@ Block events are stored in a LinkedHashSet. The game loops through block events
...
+
+#### Navigation
+Call `this.navigation` which is an `EntityNavigation` object.
+1. recalculate the path if needed
+2. return if the mob is idle
+3. if the mob is at a valid position continue following the path
+4. if the path is finished go to the next one
+
+#### [Mob Tick](https://techmcdocs.github.io/pages/GameTick/MobTick/)
+This method is different for each mob and will be addressed for each mob specifically.
```
Comment (JoakimThorsen) [inline GameTick.md:460]: I would change this to mimic the minecraft wiki:
```
### Mob Tick
Main article: [Mob Tick](https://techmcdocs.github.io/pages/GameTick/MobTick/)  
This method is different for each mob and will be addressed for each mob specifically.
```

**Review by JoakimThorsen — COMMENTED** [review]
Some of it is written like a video essay script which I'm not the biggest fan of, but that's not really a pressing issue.
The headers in a few of these are formatted differently from the rest of the wiki. There should only be one "single-hash" heading on the top of each page, with each following heading starting at two hashes.
Other comments are written inside this review

File: GameMechanics/MobSpawning.md (hunk 3 above)
Comment (Dainternetdude) [reply GameMechanics/MobSpawning.md]: changed it to use version blocks

File: GameMechanics/MobSpawning.md (hunk 2 above)
Comment (Dainternetdude) [reply GameMechanics/MobSpawning.md]: true

File: GameTick/Weather.md (hunk 1 above)
Comment (Dainternetdude) [reply GameTick/Weather.md]: oops

File: GameMechanics/MobSpawning.md (hunk 3 above)
Comment (JoakimThorsen) [reply GameMechanics/MobSpawning.md]: the change was reverted in 1.13, not 1.14 as far as I can tell

Comment (Dainternetdude) [reply GameMechanics/MobSpawning.md]: sounds good my testing agrees

**Comment by JoakimThorsen** [comment]
I'll take a look in a couple days, busy this weekend

**Review by JoakimThorsen — APPROVED** [review]
Made a commit to add specificity to the spawnable blocks section. If you're happy with it then feel free to merge, everything else looks good to me.

**Pull request merged**